docarray/array/storage/milvus/backend.py

```diff
@@ -0,0 +1,342 @@
+            schema=schema,
+            using=self._connection_alias,
+            **self._config.collection_config,
+        )
+
+    def _build_index(self):
+        index_params = {
+            'metric_type': self._config.distance,
+            'index_type': self._config.index_type,
+            'params': self._config.index_params,
+        }
+        self._collection.create_index(field_name='embedding', index_params=index_params)
+
+    def _create_or_reuse_offset2id_collection(self):
+        if has_collection(
+            self._config.collection_name + '_offset2id', using=self._connection_alias
+        ):
+            return Collection(
+                self._config.collection_name + '_offset2id',
+                using=self._connection_alias,
+            )
+
+        document_id = FieldSchema(
+            name='document_id', dtype=DataType.VARCHAR, max_length=ID_VARCHAR_LEN
+        )
+        offset = FieldSchema(
+            name='offset',
+            dtype=DataType.VARCHAR,
+            max_length=OFFSET_VARCHAR_LEN,
+            is_primary=True,
+        )
+        dummy_vector = FieldSchema(
+            name='dummy_vector', dtype=DataType.FLOAT_VECTOR, dim=1
+        )
+
+        schema = CollectionSchema(
+            fields=[offset, document_id, dummy_vector],
+            description='offset2id for DocumentArray',
+        )
+
+        return Collection(
+            name=self._config.collection_name + '_offset2id',
+            schema=schema,
+            using=self._connection_alias,
+            # **self._config.collection_config,  # we probably don't want to apply the same config here
+        )
+
+    def _ensure_unique_config(
+        self,
+        config_root: dict,
+        config_subindex: dict,
+        config_joined: dict,
+        subindex_name: str,
+    ) -> dict:
+        if 'collection_name' not in config_subindex:
+            config_joined['collection_name'] = _sanitize_collection_name(
+                config_joined['collection_name'] + '_subindex_' + subindex_name
+            )
+        return config_joined
+
+    def _doc_to_milvus_payload(self, doc):
+        return self._docs_to_milvus_payload([doc])
+
+    def _docs_to_milvus_payload(self, docs: 'Iterable[Document]'):
+        extra_columns = [
+            [self._map_column(doc.tags.get(col), col_type) for doc in docs]
+            for col, col_type in self._config.columns.items()
+        ]
+        return [
+            [doc.id for doc in docs],
+            [self._map_embedding(doc.embedding) for doc in docs],
+            [doc.to_base64(**self._config.serialize_config) for doc in docs],
+            *extra_columns,
+        ]
+
+    @staticmethod
+    def _docs_from_query_response(response):
+        return DocumentArray([Document.from_base64(d['serialized']) for d in response])
+
+    @staticmethod
+    def _docs_from_search_response(
+        responses,
+    ) -> 'Union[List[DocumentArray], DocumentArray]':
```

Comment:
I think autoboxing is done in parent Find mixin. In backends the return type is always List[DocumentArray]